src/dotty/tools/dotc/core/Types.scala

```diff
@@ -3422,8 +3422,9 @@ object Types {
         case tp: HKApply =>
           def mapArg(arg: Type, tparam: TypeParamInfo): Type = {
             val saved = variance
-            if (tparam.paramVariance < 0) variance = -variance
-            else if (tparam.paramVariance == 0) variance = 0
+            val pvariance = tparam.paramVariance
+            if (pvariance < 0) variance = -variance
+            else if (pvariance == 0) variance = 0
```

Comment:
`variance *= tparam.paramVariance` instead of the `if/else` ? (Same in the case below)